### PR TITLE
Add and configure postgres

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Start the process to roll out postgres by installing pg gems and constructing a migration.

This is pretty much the bare minimum we can do to postgresify, and is intended to separate out the devopsy stuff from actual app work. Everything in here is basically a noop - it adds a postgres instance and a table that is not set up to get used yet. So pretty safe, from a risk perspective.

I added a hobby postgres db to test this on in sandbox. I am not sure how to do the db:create / db:migrate as part of deployment, or if we _should_ do that. Likely when we roll this out for real it should be on grown-up paid dbs though.

This pull request makes the following changes:
* install and configure pg in code
* set up a table (papertrail's `versions` table) to prove migration works
* configure CI/docker

no view changes

It relates to the following issue #s: 
* Bumps #2072

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: Gemfile
Comment (colinxfleming) [inline Gemfile:29]: If we don't remove this, pg tries to run migrations twice. This should be fine since we never actually wrote a mongo migration.

File: config/initializers/devise.rb
Comment (colinxfleming) [inline config/initializers/devise.rb:266]: This forces a setting to avoid some truly bizarre pg-specific devise behavior - seems like a couple of the modules have completely different logic when using activerecord and not.

**Review by colinxfleming — COMMENTED** [review]
couple of notes

**Comment by colinxfleming** [comment]
promoting to ready for review!

File: docs/SETUP.md
Comment (lomky) [inline docs/SETUP.md:67]:
```suggestion
* Install Postgres
  * MacOS: `brew install postgres`
  * Ubuntu: `sudo apt-get -y install postgresql`, `sudo apt-get -y install libpq-dev`, `sudo apt-get -y install postgresql-client`
  * Other [linux instructions](https://www.postgresql.org/download/) (server and developer libraries needed)
```

**Comment by colinxfleming** [comment]
I rebased this to be sure it wouldn't get too far out of sync - still looking good.

A question we should resolve before we merge this - should we be doing UUIDs in here rather than serial ints?

File: Gemfile
Comment (xmunoz) [inline Gemfile:27]: Shouldn't this also be removed?

**Review by xmunoz — COMMENTED** [review]
I still see a lot of mongo packages in the `Gemfile`. What exactly is the roll out plan for this?

**Review by xmunoz — APPROVED** [review]

**Comment by lomky** [comment]
@xmunoz did you get the migration to run locally? I'm having issues but think it might be that I haven't configured postgres properly

**Comment by colinxfleming** [comment]
Okay, uuids are turning out to be a big nevermind as standard issue. I'm going to keep pgcrypto enabled but revert the rest of it.

**Comment by colinxfleming** [comment]
Okay, here's my plan:

* merge this (not at 2am though)
* deploy to stage and get that configured; if possible, figure out a way to autorun migrate via buildpack or whatever
* spin up pg:hobby (paid heroku instances) on all our instances
* deploy to funds with same configuration as stage
* probably bump our heroku app.json to spin up a pg instance too for a bit

**Review by xmunoz — CHANGES_REQUESTED** [review]
Sorry, I have to un-approve this. I can't get this build step working locally:

```
user@athena:~/workspace/dcaf_case_management$ docker-compose run --rm web rake db:seed
Starting dcaf_case_management_db_1 ... done
Starting dcaf_case_management_pg_1 ... done
warning: parser/current is loading parser/ruby27, which recognizes
warning: 2.7.1-compliant syntax, but you are running 2.7.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
rake aborted!
ActiveRecord::NoDatabaseError: FATAL:  database "dcaf_case_management_development" does not exist
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql_adapter.rb:50:in `rescue in postgresql_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql_adapter.rb:33:in `postgresql_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:887:in `new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:931:in `checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:910:in `try_to_checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:871:in `acquire_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:593:in `checkout'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:437:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:1119:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_handling.rb:221:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_handling.rb:189:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:261:in `block (3 levels) in <main>'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:258:in `each'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:258:in `flat_map'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:258:in `block (2 levels) in <main>'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:330:in `block (2 levels) in <main>'

Caused by:
PG::ConnectionBad: FATAL:  database "dcaf_case_management_development" does not exist
/usr/local/bundle/gems/pg-1.2.3/lib/pg.rb:58:in `initialize'
/usr/local/bundle/gems/pg-1.2.3/lib/pg.rb:58:in `new'
/usr/local/bundle/gems/pg-1.2.3/lib/pg.rb:58:in `connect'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql_adapter.rb:46:in `postgresql_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:887:in `new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:931:in `checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:910:in `try_to_checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:871:in `acquire_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:593:in `checkout'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:437:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:1119:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_handling.rb:221:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/connection_handling.rb:189:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:261:in `block (3 levels) in <main>'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:258:in `each'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:258:in `flat_map'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:258:in `block (2 levels) in <main>'
/usr/local/bundle/gems/activerecord-6.0.3.4/lib/active_record/railties/databases.rake:330:in `block (2 levels) in <main>'
Tasks: TOP => db:abort_if_pending_migrations
(See full trace by running task with --trace)
```

**Comment by xmunoz** [comment]
> @xmunoz did you get the migration to run locally? I'm having issues but think it might be that I haven't configured postgres properly

Nope, I did not. We might consider adding a Github Action that brings up a dev environment to test every PR.

**Comment by colinxfleming** [comment]
@xmunoz that's an error I associate with not having run `rails db:create db:migrate` - e.g. seeds can't populate into the database because the postgres isn't set up and migrated. Is `docker-compose run --rm web rails db:create db:migrate` working right, and then yielding the seed error?

File: docs/SETUP.md
Comment (xmunoz) [inline docs/SETUP.md:71]: Can you add the `docker-compose` version of these instructions above (line 20).

**Comment by xmunoz** [comment]
> @xmunoz that's an error I associate with not having run `rails db:create db:migrate` - e.g. seeds can't populate into the database because the postgres isn't set up and migrated. Is `docker-compose run --rm web rails db:create db:migrate` working right, and then yielding the seed error?

That fixed the error, and I was able to bring up a full local environment. Just make the documentation change and this should be good to go.

**Comment by xmunoz** [comment]
Oh also, do you have a plan to address the failing checks? Given that this is an intermediary change, it's probably fine that coverage decreases temporarily until the full transition is implemented.

File: docs/SETUP.md
Comment (colinxfleming) [reply docs/SETUP.md:71]: done!

**Comment by colinxfleming** [comment]
yeah, I'm not sweating the coverage drop here (-0.16%?) - makes sense that it would happen given the nature of this PR, but I can address that if you like.

Re-requested a review from you in light of the docker-compose docs change - great catch on that @xmunoz , thank you very much.

**Comment by colinxfleming** [comment]
(The coverage flag is codecov being sad about `application_record.rb` not being tested, which does not feel worth resolving to me.)

**Review by xmunoz — APPROVED** [review]

**Comment by colinxfleming** [comment]
I'm gonna ship DARIA updates late tonight to clear out the queue and then probably get this deployed tomorrow (pending an approval review from either @xmunoz or @lomky ). I like where this is winding up quite a bit tbh.

**Pull request merged**
